### DISCUSSION TITLE
nav/ahua

### DISCUSSION THREAD

**Discussion by giscus** [Yiov/vitepress-doc#15] [General]
# nav/ahua

我的vitpress文档教程

https://vitepress.yiov.top/nav/ahua.html

**Comment by supperun** [comment]
(*^▽^*)

**Comment by abs612** [comment]
大佬知道，vitepress如何处理html文件吗？可以让静态html生成一个网页，通过url就能访问到

**Reply by Yiov** [reply]
封装成 [vue组件](https://vitepress.yiov.top/components.html) 就好了，依次放入

**Reply by abs612** [reply]
> 封装成 [vue组件](https://vitepress.yiov.top/components.html) 就好了，依次放入

谢谢佬回答，看来不行了。我的是导图软件导出的html，封装组件，看来不可行。

**Comment by XingJi-love** [comment]
佬，我没有部署成功，下面这两段程序一直报错，提示路径不正确
import { NAV_DATA } from '/.vitepress/theme/untils/data'
import type { NavData } from './types'

**Reply by Yiov** [reply]
那你肯定是路径有误，仔细看一下

**Reply by XingJi-love** [reply]
佬，我是按照你的步骤来的，是不是我选择的是js的原因

**Reply by Yiov** [reply]
ahua的导航教程已经没有了，你参考也只有茂茂的了，虽然步骤都差不多

你这个不看到文件完全不知道，你贴一个demo我看看

文件data.ts和types.ts是否存在和正确

**Reply by XingJi-love** [reply]
佬，新建布局放进去就红了，好像是js的原因
![屏幕截图 2024-12-19 131806](https://github.com/user-attachments/assets/9e8326cb-8e3b-4772-9b46-c02835627cc4)

**Reply by XingJi-love** [reply]
```js
  Layout: () => {
    const props: Record<string, any> = {} //这段报红
    // 获取 frontmatter
    const { frontmatter } = useData()

    /* 添加自定义 class */
    if (frontmatter.value?.layoutClass) {
      props.class = frontmatter.value.layoutClass
    }

    return h(DefaultTheme.Layout, props)
  },
```

佬，我换成下面这样就不报红了，可以这样吗？
```js
Layout() {
    const props = {};
    const { frontmatter } = useData();

    if (frontmatter.value?.layoutClass) {
      props.class = frontmatter.value.layoutClass;
    }

    return h(DefaultTheme.Layout, props);
  },
```

**Reply by Yiov** [reply]
layout冲突了，你这都有3个了，仅留一个，重新梳理一下

可以参考我仓库的源代码

**Reply by XingJi-love** [reply]
佬，还是报错
![image](https://github.com/user-attachments/assets/f2de1e3c-7db4-4a32-95f1-edbe73b71030)

**Reply by Yiov** [reply]
哦，这个你要换成ts文件才可以

**Comment by wkwbk** [comment]
https://vitepress.yiov.top/nav/#%E6%96%B0%E5%BB%BA%E7%BB%84%E4%BB%B6

MNavLink.vue：`import { NavLink } from '../untils/types'`

MNavLinks.vue：`import type { NavLink } from '../untils/types'`

这两个文件中的这两行路径有误好像，`untils` 应该是 `utils`

**Reply by Yiov** [reply]
是的，感谢指正，文章打漏字了，请直接改文件夹名为untils

**Reply by wkwbk** [reply]
好的，还有一个问题：

```css
/* .vitepress/theme/style/nav.css */
.m-nav-layout {
  .medium-zoom-overlay,.medium-zoom-image {
    z-index: none;
  }
}
```

添加之后没有效果，F12 提示 `z-index: none;` 属性值无效

**Reply by Yiov** [reply]
本意就是让这个属性无效的，如果你有图标被穿透的情况就添加，没有就不加

**Reply by wkwbk** [reply]
我添加之后图标还是穿透的

**Reply by wkwbk** [reply]
麻烦您帮我看下可以吗

https://github.com/wkwbk/wkwbk.github.io/blob/main/docs/.vitepress/theme/style/nav.css

**Reply by Yiov** [reply]
改成这样呢 z-index: 0 !important;

**Reply by wkwbk** [reply]
改成这样没有穿透了，感谢

**Comment by XingJi-love** [comment]
佬，部署后页面就显示这么大，不知道为啥，该配置的都配置了，仓库地址: https://github.com/XingJi-love/Blog-VitePress
![屏幕截图 2025-01-10 135319](https://github.com/user-attachments/assets/31042fe3-2b45-47da-be48-f7e81884a06e)
不像你的这么大
![image](https://github.com/user-attachments/assets/c389affd-839b-42eb-99bd-48d91fe9b1ba)

**Reply by Yiov** [reply]
布局的配置没有生效，修改 `theme/index.ts` 配置

```diff
export default {

    Layout() {
        const props: Record<string, any> = {}
        // 获取 frontmatter
        const { frontmatter } = useData()

        /* 添加自定义 class */
        if (frontmatter.value?.layoutClass) {
            props.class = frontmatter.value.layoutClass
        }

-        return h(MyLayout, null, {
-        default: () => h(DefaultTheme.Layout, props, {
-            'doc-footer-before': () => h(backtotop), // 使用doc-footer-before插槽
-           })
-        })

+        return h(MyLayout, props)
    },
}
```

至于 `backtotop` 组件可以放到 `MyLayout` 中 即可

**Reply by XingJi-love** [reply]
谢谢佬，解决了

**Comment by zxyang3636** [comment]
大佬这里应该怎么改呀，如果这么写，那切换黑白主题时，就没有丝滑过度了，怎么才能把他俩整个到一起呀：
```
import MyLayout from "./components/MyLayout.vue";
import xgplayer from "./components/xgplayer.vue";
import { h } from "vue";
import { useData } from "vitepress";
import MNavLinks from "./components/MNavLinks.vue";

export default {
  extends: DefaultTheme,
  Layout: () => {
    const props: Record<string, any> = {};
    // 获取 frontmatter
    const { frontmatter } = useData();

    /* 添加自定义 class */
    if (frontmatter.value?.layoutClass) {
      props.class = frontmatter.value.layoutClass;
    }

    return h(DefaultTheme.Layout, props);
  },
```

**Reply by Yiov** [reply]
你可以看一下我仓库的代码，先写自定义视图过度组件，再把其他组件写入这个组件中即可
